frontend/dialob-composer-material/src/dialob/react/useComposer.ts

```diff
@@ -3,145 +3,153 @@ import { ComposerContext } from './ComposerContext';
-    dispatch({type: 'deleteLanguage', language});
-  }
-
-  return {
-    addItem,
-    updateItem,
-    updateLocalizedString,
-    changeItemType,
-    deleteItem,
-    setItemProp,
-    deleteItemProp,
-    moveItem,
-    createValidation,
-    setValidationMessage,
-    setValidationExpression,
-    deleteValidation,
-    createValueSet,
-    setValueSetEntries,
-    addValueSetEntry,
-    updateValueSetEntry,
-    deleteValueSetEntry,
-    moveValueSetEntry,
-    setGlobalValueSetName,
-    deleteGlobalValueSet,
-    setMetadataValue,
-    createVariable,
-    updateContextVariable,
-    updateExpressionVariable,
-    deleteVariable,
-    addLanguage,
-    deleteLanguage,
-    form: state
-  };
-
-}
\ No newline at end of file
+	const changeItemType = (itemId: string, config: DialobItemTemplate) => {
+		dispatch({ type: 'changeItemType', itemId, config });
+	};
+
+	const deleteItem = (itemId: string) => {
+		dispatch({ type: 'deleteItem', itemId });
+	};
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	const setItemProp = (itemId: string, key: string, value: any) => {
+		dispatch({ type: 'setItemProp', itemId, key, value });
+	}
+
+	const deleteItemProp = (itemId: string, key: string) => {
+		dispatch({ type: 'deleteItemProp', itemId, key });
+	}
+
+	const moveItem = (itemId: string, fromIndex: number, toIndex: number, fromParent: string, toParent: string) => {
+		dispatch({ type: 'moveItem', itemId, fromIndex, toIndex, fromParent, toParent });
+	}
+
+	const createValidation = (itemId: string, rule?: ValidationRule) => {
+		dispatch({ type: 'createValidation', itemId, rule });
+	}
+
+	const setValidationMessage = (itemId: string, index: number, language: string, message: string) => {
+		dispatch({ type: 'setValidationMessage', itemId, index, language, message });
+	}
+
+	const setValidationExpression = (itemId: string, index: number, expression: string) => {
+		dispatch({ type: 'setValidationExpression', itemId, index, expression });
+	}
+
+	const deleteValidation = (itemId: string, index: number) => {
+		dispatch({ type: 'deleteValidation', itemId, index });
+	}
+
+	const createValueSet = (itemId: string | null, entries?: ValueSetEntry[]) => {
+		dispatch({ type: 'createValueSet', itemId, entries });
+	};
+
+	const setValueSetEntries = (valueSetId: string, entries: ValueSetEntry[]) => {
+		dispatch({ type: 'setValueSetEntries', valueSetId, entries });
+	}
+
+	const addValueSetEntry = (valueSetId: string, entry?: ValueSetEntry) => {
+		dispatch({ type: 'addValueSetEntry', valueSetId, entry });
+	}
+
+	const updateValueSetEntry = (valueSetId: string, index: number, entry: ValueSetEntry) => {
+		dispatch({ type: 'updateValueSetEntry', valueSetId, index, entry });
+	}
+
+	const updateValueSetEntryOld = (valueSetId: string, index: number, valueSetEntryId: string | undefined, text: string, language: string) => {
+		dispatch({ type: 'updateValueSetEntryOld', valueSetId, index, valueSetEntryId, text, language });
```

Comment:
"...Old" is not a good name as it suggests something deprecated or obsolete. please propose a better name